mlflow/utils/autologging_utils.py

```diff
@@ -119,6 +125,7 @@ def wrap_patch(destination, name, patch, settings=None):
 
     original = getattr(destination, name)
     wrapped = functools.wraps(original)(patch)
+    wrapped.__signature__ = inspect.signature(original)
```

Comment:
`functools.wraps()` alone doesn't preserve the original signature of the function. For example, invoking `inspect.getargspec()` on a wrapped function yields the wrapper's signature, rather than the underlying function's signature. This may cause problems for frameworks that perform argument inspection; this concern has been raised by stakeholders.